lib/builders/MaterialBuilder.ts

```diff
@@ -73,7 +73,12 @@ export class MaterialBuilder {
    * Outputs a static theme stylesheet, based on the specified options
    */
   _buildTheme(buildData: MaterialToolsData): MaterialToolsOutput {
-    if (!this._themeBuilder) {
+
+    // When the ThemeBuilder is not initialized and theme definitions are specified.
+    if (!this._themeBuilder && this._themes) {
+      let moduleName = path.join(buildData.package.module, this._options.mainFilename);
+      this._themeBuilder = new ThemeBuilder(this._themes, this._options.palettes, moduleName);
+    } else if (!this._themeBuilder) {

```

Comment:
In what case wouldn't we have a themeBuilder at this point? I think we can shuffle things around a bit by removing the `else` here and having a `if (!this._options.themes) return` at the top and then only initialising the theme builder conditionally.